apps/omg_watcher_rpc/lib/web/validators/block_constraints.ex

```diff
@@ -0,0 +1,40 @@
+# Copyright 2019 OmiseGO Pte Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+defmodule OMG.WatcherRPC.Web.Validator.BlockConstraints do
+  @moduledoc """
+  Validates `/block.all` query parameters
+  """
+
+  import OMG.Utils.HttpRPC.Validator.Base
+
+  @doc """
+  Validates possible query constraints, stops on first error.
+  """
+  @spec parse(%{binary() => any()}) :: {:ok, Keyword.t()} | {:error, any()}
+  def parse(params) do
+    constraints = [
+      limit: [:pos_integer, :optional],
+      page: [:pos_integer, :optional]
+    ]
+
+    Enum.reduce_while(constraints, {:ok, []}, fn {constraint, validators}, {:ok, list} ->
+      case expect(params, Atom.to_string(constraint), validators) do
```

Comment:
it seems like you really need string key tuples. whats the other reason they're defined with atoms?

Reply:
Hmm it's being parsed downstream as atoms. But I refactored to remove this and do `String.to_existing_atom/1` only on happy path instead.

Looks like a good target for non-Ecto changeset? For later PR.